### PR TITLE
GRD: Try to find local idea/clion installed by toolbox

### PR DESCRIPTION
**Pull request by ice1000**
Signed-off-by: ice1000 <ice1000kotlin@foxmail.com>

So in most cases people won't need to wait for gradle download a giant dependency.

macOS is not supported because I don't have a mac and I have no idea where toolbox will install your IDE.
I can add macOS if anyone can tell me where the default installation path is.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -266,6 +270,7 @@ project(":idea") {
 project(":clion") {
     intellij {
         version = clionVersion
+        alternativeIdePath = clionLocalPath
```
Comment (Undin) [inline build.gradle.kts]: It affects nothing because this property makes sense only for root module

Comment (ice1000) [reply build.gradle.kts]: Ah, I didn't know this. I thought that was for the `runIde` action.

File: build.gradle.kts [hunk 2]
```diff
@@ -526,3 +532,25 @@ fun List<String>.execute(wd: String? = null, ignoreExitCode: Boolean = false): S
...
+        os.startsWith("Windows") -> "C:\\Users\\$user\\AppData\\Local\\JetBrains\\Toolbox\\apps"
+        os == "Linux" -> "/home/$user/.local/share/JetBrains/Toolbox/apps"
+        else -> return null
+    }
+    return file(root)
+        .resolve(ide)
+        .resolve("ch-0")
+        .listFiles()
+        .orEmpty()
+        .filter { it.isDirectory && it.name.startsWith(expectedMajorVersion) }
```
[the remark this answers is not in the document]
Comment (ice1000) [reply build.gradle.kts]: That parameter is supposed to be assigned as `183`, `191`, etc

**Comment by ice1000** [comment]
@Undin the reason looks quite reasonable to me. Thanks for your explanation!

**Pull request closed without merging**